usefulUtilities/hiFiCalendar/entityScripts/hiFiMeetingZone.js

```diff
@@ -7,30 +7,45 @@
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 
 (function() {
-    var ORIGIN = {x:0, y:0, z: 0};
-    var RADIUS = 100000;
     
     var _this;
-    var roomName;
-    var occupantTextEntityId;
 
-    var MeetingZone = function() {
-        _this = this;        
+    var MeetingZone = function() {  
+        _this = this;     
     };
 
     MeetingZone.prototype = {
-        preload: function(entityId) {
-            _this.entityId = entityId;
-            roomName = Entities.getEntityProperties(_this.entityId, ['name']).name;
-            occupantTextEntityId = Entities.findEntitiesByName(roomName + '_OCCUPANTS', ORIGIN, RADIUS)[0];
+        preload: function(entityID) {
```

Comment:
on preload, you need to check if the avatar that loads this script is standing within the boundaries of the zone, and add them to the list of users inside the zone if so.

i've written an example of a function that does this below
```
    var HALF = 0.5;
    function positionIsInsideEntityBounds(entityID, targetPosition) {
        targetPosition = targetPosition || MyAvatar.position;

        var properties = Entities.getEntityProperties(entityID, ["position", "dimensions", "rotation"]);
        var entityPosition = properties.position;
        var entityDimensions = properties.dimensions;
        var entityRotation = properties.rotation;
        
        var worldOffset = Vec3.subtract(targetPosition, entityPosition);
        targetPosition = Vec3.multiplyQbyV(Quat.inverse(entityRotation), worldOffset);

        var minX = -entityDimensions.x * HALF;
        var maxX = entityDimensions.x * HALF;
        var minY = -entityDimensions.y * HALF;
        var maxY = entityDimensions.y * HALF;
        var minZ = -entityDimensions.z * HALF;
        var maxZ = entityDimensions.z * HALF;

        return (targetPosition.x >= minX && targetPosition.x <= maxX
            && targetPosition.y >= minY && targetPosition.y <= maxY
            && targetPosition.z >= minZ && targetPosition.z <= maxZ);
    }
```

Reply:
When I reload the script with people in the room they get picked up, so I'm not sure this is necessary.